packages/compatibility/backend/sirius-components-compatibility/src/main/java/org/eclipse/sirius/components/compatibility/diagrams/AbstractNodeMappingChilrenLayoutStrategyProvider.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.compatibility.diagrams;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+import org.eclipse.sirius.components.diagrams.FreeFormLayoutStrategy;
+import org.eclipse.sirius.components.diagrams.ILayoutStrategy;
+import org.eclipse.sirius.components.diagrams.LayoutDirection;
+import org.eclipse.sirius.components.diagrams.ListLayoutStrategy;
+import org.eclipse.sirius.components.representations.VariableManager;
+import org.eclipse.sirius.diagram.business.api.query.ContainerMappingQuery;
+import org.eclipse.sirius.diagram.description.AbstractNodeMapping;
+import org.eclipse.sirius.diagram.description.ContainerMapping;
+
+/**
+ * Used to compute the children layout strategy using the definition of an abstract node mapping.
+ *
+ * @author gcoutable
+ */
+public class AbstractNodeMappingChilrenLayoutStrategyProvider implements Function<VariableManager, ILayoutStrategy> {
```

Comment:
`AbstractNodeMappingChildrenLayoutSTrategyProvider` is now used. It will return a free form layout strategy if the option is explicitly defined in the VSM and return a list layout strategy only for vertical stack.